cmd/zfs/zfs_main.c

```diff
@@ -7289,6 +7389,230 @@ zfs_do_change_key(int argc, char **argv)
 	return (0);
 }
 
+/*
+ * 1) zfs project -p <id> <file|directory ...>
+ *    Set project ID with the given value for the specified file or directory.
+ *
+ * 2) zfs project -c [-d|-r [-p id] [-0]] <file|directory ...>
+ *    List the file's or directory children's project ID (and the inherit
+ *    flag if have).
+ *    -d: Show the directory's itself project ID (and inherit flag if has).
+ *    -r: Check the tree quota status. If all are fine, report nothing;
+ *        otherwise, the files and directories with inconsistent project IDs
+ *        or without inherit flag will be reported.
+ *    -p: Specify the expected project ID for checking tree quota. It needs
+ *        to be used together with "-r".
+ *    -0: Print entries found by -c with a trailing NUL instead of a newline,
+ *        like "find -print0". It needs to be specified together with "-r".
+ *
+ * 3) zfs project -C [-r] [-k] <file|directory ...>
+ *    Clear project ID inherit flag for the given file or directory.
+ *    -r: All descendants' (of the directory) inherit flag will be cleared.
+ *    -k: Keep the porject ID unchanged. If not specified, the directory's
+ *        (and all its descendants' for "-r" specified) project ID will be
+ *        reset as zero.
+ *
+ * 4) zfs project -s [-r] [-p id] <file|directory ...>
+ *    Set project ID inherit flag for the given file or directory.
```

Comment:
Setting inherit flag on regular file has not too much functionality, but it can simply the logic to avoid distinguishing file mode when operate tree quota. On the other hand, XFS also does NOT distinguish that.